### PR TITLE
Godoc for kubelet pod manager

### PR DESCRIPTION
**Pull request by pmorie**
Improve godoc for pod manager and mirror client.

@kubernetes/sig-node

### CODE REVIEW

File: pkg/kubelet/pod/manager.go [hunk 1]
```diff
@@ -24,49 +24,81 @@ import (
...
 	GetPodByName(namespace, name string) (*api.Pod, bool)
+	// GetPodByUID provides the (non-mirror) pod that matches pod UID, as well as
+	// whether the pod is found.
 	GetPodByUID(types.UID) (*api.Pod, bool)
+
+	// GetPodByMirrorPod returns the static pod for the given mirror pod and
+	// whether it was known to the pod manger.
 	GetPodByMirrorPod(*api.Pod) (*api.Pod, bool)
+	// GetMirrorPodByPod returns the mirror pod for the given static pod and
+	// whether it was known to the pod manager
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: nit, missing the tailing period.

File: pkg/kubelet/pod/manager.go [hunk 2]
```diff
@@ -24,49 +24,81 @@ import (
...
 	// SetPods replaces the internal pods with the new pods.
 	// It is currently only used for testing.
 	SetPods(pods []*api.Pod)
 
-	// Methods that modify a single pod.
+	// AddPod adds the given pod to the manager.
 	AddPod(pod *api.Pod)
+	// UpdatePod updates the given pod in the manager.
 	UpdatePod(pod *api.Pod)
+	// DeletePod deletes the given pod from the manager.  For mirror pods,
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: nit, extra spaces after period?

File: pkg/kubelet/pod/manager.go [hunk 3]
```diff
@@ -24,49 +24,81 @@ import (
...
+	// this means deleting the mappings related to mirror pods.  For non-
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: ditto

File: pkg/kubelet/pod/manager.go [hunk 4]
```diff
@@ -24,49 +24,81 @@ import (
...
+	// mirror pods, this means deleting from indexes for all pods.
 	DeletePod(pod *api.Pod)
 
+	// DeleteOrphanedMirrorPods deletes all mirror pods which do not have
+	// associated static pods. This method sends deletion requets to the API
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: s/requets/requests

File: pkg/kubelet/pod/manager.go [hunk 5]
```diff
@@ -24,49 +24,81 @@ import (
...
+	// server, but does NOT modify the internal pod storage in basicManager.
 	DeleteOrphanedMirrorPods()
+	// TranslatePodUID returns the UID which is the mirror pod or static pod of
+	// the pod with the given UID.  If the UID belongs to a mirror pod, maps it to
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: nit, extra space after period (not sure if it's your intention to keep that?)

File: pkg/kubelet/pod/manager.go [hunk 6]
```diff
@@ -24,49 +24,81 @@ import (
...
+	// the UID of its static pod. Otherwise, return the original UID.
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: s/return/returns

File: pkg/kubelet/pod/manager.go [hunk 7]
```diff
@@ -24,49 +24,81 @@ import (
...
+	// GetPodByName returns the (non-mirror) pod that matches full name, as well as
+	// whether the pod was found.
 	GetPodByFullName(podFullName string) (*api.Pod, bool)
+	// GetPodByName provides the (non-mirror) pod that matches namespace and
+	// name, as well as whether the pod was found.
 	GetPodByName(namespace, name string) (*api.Pod, bool)
+	// GetPodByUID provides the (non-mirror) pod that matches pod UID, as well as
+	// whether the pod is found.
 	GetPodByUID(types.UID) (*api.Pod, bool)
+
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: Are we going to keep empty lines between functions?

File: pkg/kubelet/pod/manager.go [hunk 8]
```diff
@@ -24,49 +24,81 @@ import (
...
+	// the UID of its static pod. Otherwise, return the original UID.
+	//
+	// All public-facing functions should perform this translation for UIDs
+	// because user may provide a mirror pod UID, which is not recognized by
+	// internal Kubelet functions.
 	TranslatePodUID(uid types.UID) types.UID
+	// GetUIDTranslations returns the mappings of static pod UIDs to mirror pod
+	// UIDs and mirror pod UIDs to static pod UIDs.
 	GetUIDTranslations() (podToMirror, mirrorToPod map[types.UID]types.UID)
+	// IsMirrorPodOf return true if mirrorPod is a correct representation of
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: s/return/returns

File: pkg/kubelet/pod/manager.go [hunk 9]
```diff
@@ -118,6 +151,9 @@ func (pm *basicManager) UpdatePod(pod *api.Pod) {
 	pm.updatePodsInternal(pod)
 }
 
+// updatePodsInternal replaces the given pods in the current state of the
+// manager, updating the various indices.  The caller is assumed to hold the
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: nit, same question on the extra space.

File: pkg/kubelet/pod/mirror_client.go [hunk 10]
```diff
@@ -25,23 +25,30 @@ import (
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
 )
 
-// Mirror client is used to create/delete a mirror pod.
+// MirrorClient knows how create/delete a mirror pod in the API server.
```
Comment (yifan-gu) [inline pkg/kubelet/pod/mirror_client.go]: s/how/how to

File: pkg/kubelet/pod/mirror_client.go [hunk 11]
```diff
@@ -25,23 +25,30 @@ import (
...
+	// the static pod.
+	CreateMirrorPod(pod *api.Pod) error
+	// DeleteMirrorPod deletes the mirror pod with the given full name from
+	// the API server or returns an error.
+	DeleteMirrorPod(podFullName string) error
 }
 
+// basicMirrorClient is a functional MirrorClient.
 type basicMirrorClient struct {
 	// mirror pods are stored in the kubelet directly because they need to be
```
Comment (yifan-gu) [inline pkg/kubelet/pod/mirror_client.go]: s/mirror/Mirror

File: pkg/kubelet/pod/mirror_client.go [hunk 12]
```diff
@@ -25,23 +25,30 @@ import (
 	kubetypes "k8s.io/kubernetes/pkg/kubelet/types"
 )
 
-// Mirror client is used to create/delete a mirror pod.
+// MirrorClient knows how create/delete a mirror pod in the API server.
 type MirrorClient interface {
-	CreateMirrorPod(*api.Pod) error
-	DeleteMirrorPod(string) error
+	// CreateMirrorPod creates a mirror pod in the API server for the given
+	// pod or returns an error.  The mirror pod will have the same annotations
```
Comment (yifan-gu) [inline pkg/kubelet/pod/mirror_client.go]: nit, same question on the extra space.

**Comment by yifan-gu** [comment]
Thanks @pmorie for the clean up! :+1: :heart:

File: pkg/kubelet/pod/manager.go (hunk 2 above)
Comment (pmorie) [inline pkg/kubelet/pod/manager.go]: All my comment sin kubernetes use two spaces after the period :)

File: pkg/kubelet/pod/manager.go [hunk 13]
```diff
@@ -24,49 +24,81 @@ import (
...
 	SetPods(pods []*api.Pod)
 
-	// Methods that modify a single pod.
+	// AddPod adds the given pod to the manager.
 	AddPod(pod *api.Pod)
+	// UpdatePod updates the given pod in the manager.
 	UpdatePod(pod *api.Pod)
+	// DeletePod deletes the given pod from the manager.  For mirror pods,
+	// this means deleting the mappings related to mirror pods.  For non-
+	// mirror pods, this means deleting from indexes for all pods.
```
Comment (Random-Liu) [inline pkg/kubelet/pod/manager.go]: all non-mirror pods?

**Comment by Random-Liu** [comment]
@pmorie LGTM with a little nit.

File: pkg/kubelet/pod/manager.go (hunk 13 above)
Comment (pmorie) [inline pkg/kubelet/pod/manager.go]: Yeah, that is probably better phrasing.

**Comment by pmorie** [comment]
@Random-Liu @yifan-gu PTAL

File: pkg/kubelet/pod/manager.go [hunk 14]
```diff
@@ -24,49 +24,79 @@ import (
...
 // A mirror pod has the same pod full name (name and namespace) as its static
 // counterpart (albeit different metadata such as UID, etc). By leveraging the
-// fact that kubelet reports the pod status using the pod full name, the status
-// of the mirror pod always reflects the actual status of the static pod.
-// When a static pod gets deleted, the associated orphaned mirror pod will
-// also be removed.
-
+// fact that the kubelet reports the pod status using the pod full name, the
+// status of the mirror pod always reflects the actual status of the static
+// pod. When a static pod gets deleted, the associated orphaned mirror pod
```
Comment (yifan-gu) [inline pkg/kubelet/pod/manager.go]: Wanna two spaces after period.

**Comment by yifan-gu** [comment]
LGTM

File: pkg/kubelet/pod/manager.go (hunk 14 above)
Comment (Random-Liu) [inline pkg/kubelet/pod/manager.go]: nit, there are also some `. ` in front. :)

**Comment by Random-Liu** [comment]
@pmorie LGTM with a single nit.

**Comment by pmorie** [comment]
@Random-Liu @yifan-gu Personally I don't think number of spaces between comment sentences is worth the time it'll take to change them.  Do you disagree?

**Comment by yifan-gu** [comment]
@pmorie Yes, I am just nit picking :) 
LGTM

**Comment by goltermann** [comment]
is this an e2e-not-required candidate?

**Comment by pmorie** [comment]
@goltermann yep

**Pull request merged**